### ISSUE TITLE
PasswordUtils blanks out the password causing the AuthKeyboardInteractive to always fail

### ISSUE THREAD

**Issue by kgston** [hierynomus/sshj#677]
The `authPassword` method takes the same instance of the `PasswordFinder` into both `AuthMethod`. If for some reason the first `AuthMethod` fails (`password` method in this case, e.g. not an accepted method on the host) and it then attempts to authenticate via the `keyboard-interactive` method, the password would have been cleared out after it was retrieved on the first attempt causing the second attempt to always fail.

My current workaround is to create both `AuthMethod` with separate instances of `PasswordFinder` with the same password and that works for me, but I'm assuming the password is being cleared for security reasons? Creating 2 instances of the password and potentially only using the first once will leave the second instance potentially exploitable. Not sure how you would like to address this issue.

https://github.com/hierynomus/sshj/blob/45b2f32b148e06201a028b9480a98249b6c3b446/src/main/java/net/schmizz/sshj/SSHClient.java#L286-L289

**Comment by hierynomus** [comment]
Yes, the default `PasswordFinder` ensures that the password you give in gets blanked out after a single try so that we don't keep it. You can create your own PasswordFinder that you blank out yourself after authentication is successful as it's just an interface. Or prefer to connect using private keys which is a lot safer.

**Comment by kgston** [comment]
Yes, I understand that it is easy to override the interface and create your own `PasswordFinder` implementation, but the default `authPassword` implementation would never would never work with a server that only supports `keyboard-interactive` as it is right now. Would be great if that would be fixed so that others won't need to trip over such a detail.

Alternatively, I think removing the `AuthKeyboardInteractive` `AuthMethod` from `authPassword` and documenting how to add the implementation properly along with the `PasswordFinder` blank out implicit feature, would be a viable alternative and helpful for others. Just some suggestions.
